modules/metastorage-client/src/main/java/org/apache/ignite/internal/metastorage/client/CursorImpl.java

```diff
@@ -101,10 +104,10 @@
                         cursorId -> metaStorageRaftGrpSvc.<Boolean>run(new CursorHasNextCommand(cursorId))).get();
             }
             catch (InterruptedException | ExecutionException e) {
-                if (e.getCause().getClass().equals(NodeStoppingException.class))
+                if (e.getCause() != null && e.getCause().getClass().equals(NodeStoppingException.class))
                     return false;
 
-                LOG.error("Unable to evaluate cursor hasNext command", e);
+                LOG.warn("Unable to evaluate cursor hasNext command");
```

Comment:
Why it's warn and not an error?
Why we don't propagate e to log?

Reply:
Because right after that we throw `new IgniteInternalException(e);` and exception should be handled later. In some cases exception in this method doesn't mean that there is an error, for example `InterruptedException` or `ExecutionException` with `RejectedExecutionException` as the cause.

Reply:
I've changed it to debug with the exception.